core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -37,7 +37,7 @@ round_down(uintptr_t v, uintptr_t b)
 #endif
 
 void *
-os_mmap(void *hint, size_t size, int prot, int flags)
+os_mmap(void *hint, size_t size, int prot, int flags, os_file_handle *file)
```

Comment:
Per my understanding, os_file_handle is already a handle and can be used inside the function body. And for the caller, if it wants to pass the invalid file handle, it can pass `os_get_invalid_handle()` to os_mmap: in posix, os_get_invalid_handle returns -1, and in windows, it returns NULL.

Reply:
Get it